functions/public/Invoke-WPFUIElements.ps1

```diff
@@ -0,0 +1,315 @@
+function Invoke-WPFUIElements {
+    <#
+    .SYNOPSIS
+        Adds UI elements to a specified Grid in the WinUtil GUI based on a JSON configuration.
+    .PARAMETER configVariable
+        The variable/link containing the JSON configuration.
+    .PARAMETER targetGridName
+        The name of the grid to which the UI elements should be added.
+    .PARAMETER columncount
+        The number of columns to be used in the Grid. If not provided, a default value is used based on the panel.
+    .EXAMPLE
+        Invoke-WPFUIElements -configVariable $sync.configs.applications -targetGridName "install" -columncount 5
+    .NOTES
+        Future me/contributer: If possible please wrap this into a runspace to make it load all panels at the same time.
+    #>
+
+    param(
+        [Parameter(Mandatory)]
+        [PSCustomObject]$configVariable,
+
+        [Parameter(Mandatory)]
+        [string]$targetGridName,
+
+        [Parameter(Mandatory)]
+        [int]$columncount
+    )
+
+    $theme = $sync.configs.themes.$ctttheme
+
+    # Convert PSCustomObject to Hashtable
+    $configHashtable = @{}
+    $configVariable.PSObject.Properties.Name | ForEach-Object {
+        $configHashtable[$_] = $configVariable.$_
+    }
+
+    $organizedData = @{}
+    # Iterate through JSON data and organize by panel and category
+    foreach ($entry in $configHashtable.Keys) {
+        $entryInfo = $configHashtable[$entry]
+
+        # Create an object for the application
+        $entryObject = [PSCustomObject]@{
+            Name = $entry
+            Order = $entryInfo.order
+            Category = $entryInfo.Category
+            Content = $entryInfo.Content
+            Choco = $entryInfo.choco
+            Winget = $entryInfo.winget
+            Panel = if ($entryInfo.Panel) { $entryInfo.Panel } else { "0" }
+            Link = $entryInfo.link
+            Description = $entryInfo.description
+            Type = $entryInfo.type
+            ComboItems = $entryInfo.ComboItems
+            Checked = $entryInfo.Checked
+            ButtonWidth = $entryInfo.ButtonWidth
+        }
+
+        if (-not $organizedData.ContainsKey($entryObject.Panel)) {
+            $organizedData[$entryObject.Panel] = @{}
+        }
+
+        if (-not $organizedData[$entryObject.Panel].ContainsKey($entryObject.Category)) {
+            $organizedData[$entryObject.Panel][$entryObject.Category] = @()
+        }
+
+        # Store application data in an array under the category
+        $organizedData[$entryObject.Panel][$entryObject.Category] += $entryObject
+    }
+
+    # Retrieve the main window and the target Grid by name
+    $window = $sync["Form"]
+    $targetGrid = $window.FindName($targetGridName)
```

Comment:
```suggestion
    $window = $sync["Form"]
    $targetGrid = $window.FindName($targetGridName)
    if ($targetGrid -eq $null) {
        throw [GenericException]::new("[Invoke-WPFUIElements] Failed to retrieve the target grid by name, provided name: $targetGridName")
    }
```
Add error checking, this'll make sure that anyone using this function doesn't provide an incorrect grid target name by accident, and because FindName method doesn't throw an exception upon not finding provided element... we're check the returned value if it's null or not.

Reply:
I already have the params set as mandatory, idk if this is needed in this case

Reply:
> I already have the params set as mandatory, idk if this is needed in this case

@MyDrift-user Having parameters set as mandatory doesn't mean PowerShell will check **the provided values**... so I could just pass some nonsense string to the function, and it won't complain.. only when you want to do anything with that returned grid object will PowerShell complain saying "Can't Call from Null Reference" or something along those lines.

Reply:
done